src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -67,16 +67,10 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
             var resultDic = inputSchema.Columns.ToDictionary(x => x.Name);
             foreach (var column in _columns)
             {
-                var originalColumn = inputSchema.FindColumn(column.Source);
-                if (originalColumn != null)
-                {
-                    var col = new SchemaShape.Column(column.Name, originalColumn.Kind, originalColumn.ItemType, originalColumn.IsKey, originalColumn.MetadataKinds);
-                    resultDic[column.Name] = col;
-                }
-                else
-                {
+                if (!inputSchema.TryFindColumn(column.Source, out var originalColumn))
                     throw _host.ExceptParam(nameof(inputSchema), $"{column.Source} not found in {nameof(inputSchema)}");
```

Comment:
>ExceptParam [](start = 32, length = 11)

Now that we're changing that, let's do `ExceptSchemaMismatch`

Reply:
Thanks @Zruty0! I will do this. May I ask, what is the reason for this being the correct behavior here, but incorrect in a place like, say, `TrainerEstimatorBase.cs`? (The resulting exceptions all being thrown via `Except`.) I couldn't quite figure out the system of whether to call this "schema" exception method or something else, sometimes.

---
In reply to: [215724217](https://github.com/dotnet/machinelearning/pull/845#discussion_r215724217) [](ancestors = 215724217)